lib/test/Test/Pos/Txp/Toil/UtxoSpec.hs

```diff
@@ -459,3 +467,11 @@ txShouldFailWithPlutus res err = case res of
     other -> expectationFailure $
         "expected: Left ...: " <> show (WitnessScriptError err) <> "\n" <>
         " but got: " <> show other
+
+-- | Transaction should fail with a 'ToilInvalidWitness' error.
+txShouldFailWithInvalidWitness :: Either ToilVerFailure () -> Expectation
+txShouldFailWithInvalidWitness = \case
+    Left ToilInvalidWitness{} -> pass
```

Comment:
@gromak I am kinda stuck here trying to make test fail with `ToilInvalidWitness` ...

Reply:
It'd be good to also check that `WitnessVerFailure` inside `ToilInvalidWitness` is actually `WitnessAdoptedScriptVerMismatch`.